dbptk-modules/dbptk-filter-inventory/src/main/java/com/databasepreservation/modules/inventory/InventoryFilter.java

```diff
@@ -0,0 +1,191 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+
+import com.databasepreservation.managers.ModuleConfigurationManager;
+import com.databasepreservation.model.data.Cell;
+import com.databasepreservation.model.data.Row;
+import com.databasepreservation.model.data.SimpleCell;
+import com.databasepreservation.model.exception.ModuleException;
+import com.databasepreservation.model.modules.filters.DatabaseFilterModule;
+import com.databasepreservation.model.reporters.Reporter;
+import com.databasepreservation.model.structure.ColumnStructure;
+import com.databasepreservation.model.structure.DatabaseStructure;
+import com.databasepreservation.model.structure.TableStructure;
+
+public class InventoryFilter implements DatabaseFilterModule {
+
+  private DatabaseFilterModule exportModule;
+  private DatabaseStructure databaseStructure;
+
+  private List<Integer> indexOfHeaders;
+
+  private String prefixPath;
+  private Path dirPath;
+  private final boolean printHeader;
+  private String separator;
+
+  private File csv_file;
+  private OutputStreamWriter outStreamWriter;
+  private CSVPrinter csv_printer;
+  private boolean do_export;
+
+  public InventoryFilter(String prefix, Path dirPath, boolean pPrintHeader, String separator) {
+    this.prefixPath = prefix;
+    this.dirPath = dirPath;
+    this.printHeader = pPrintHeader;
+    this.separator = separator;
+  }
+
+  @Override
+  public void initDatabase() throws ModuleException {
+    this.exportModule.initDatabase();
+  }
+
+  @Override
+  public void setIgnoredSchemas(Set<String> ignoredSchemas) {
+    this.exportModule.setIgnoredSchemas(ignoredSchemas);
+  }
+
+  @Override
+  public void handleStructure(DatabaseStructure structure) throws ModuleException {
+    databaseStructure = structure;
+    this.exportModule.handleStructure(structure);
+  }
+
+  @Override
+  public void handleDataOpenSchema(String schemaName) throws ModuleException {
+    this.exportModule.handleDataOpenSchema(schemaName);
+  }
+
+  @Override
+  public void handleDataOpenTable(String tableId) throws ModuleException {
+    do_export = false;
+    TableStructure currentTable = databaseStructure.getTableById(tableId);
+    String currentTableName = currentTable.getId();
+
+    indexOfHeaders = new ArrayList<>();
+    List<String> listOfHeaders = new ArrayList<>();
+
+    try {
+      int index = 0;
+
+      for (ColumnStructure column : currentTable.getColumns()) {
+        if (currentTable.isFromCustomView() || ModuleConfigurationManager.getInstance().getModuleConfiguration()
+          .isInventoryColumn(currentTable.getSchema(), currentTable.getName(), column.getName())) {
+          String header = currentTableName + "." + column.getName();
```

Comment:
since there is a file per table, the header could just simple be column.getName()